// Last updated: 7/16/2026, 1:12:52 AM
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        double[] merged = new double[nums1.length+ nums2.length];
        for (int i = 0; i < nums1.length; i++) {
            merged[i] = (double) nums1[i]; // Explicit cast
        }
        for (int i = 0; i < nums2.length; i++) {
            merged[nums1.length + i] = (double) nums2[i]; // Explicit cast
        }
         Arrays.sort(merged);
         int n = merged.length;
         if(n%2 !=0){
            return merged[(n-1)/2];
         }
         else { 
            return (merged[n / 2 - 1] + merged[n / 2]) / 2.0;
        }
    }
}